src/imfilter_cpu.jl

```diff
@@ -2,7 +2,8 @@
 # Licensed under the MIT License. See LICENCE in the project root.
 # ------------------------------------------------------------------
 
-function imfilter_cpu(img::AbstractArray{T,N},
-                      kern::AbstractArray{K,N}) where {T<:Real,K<:Real,N}
+function imfilter_resource(resource,
```

Comment:
I think you ignored my previous suggestion about removing type annotations? Can you please remove the types from `img` and `kern`? They don't help the compiler and are just cumbersome for anyone reading the code.

Reply:
Ok. I can do it for you. However, it is worth mentioning that the original code of **imfilter_gpu** and **imfilter_cpu**, which were renamed to **imfilter_resource**, has these type annotations. I only included the resource parameter. I was avoiding changing the original code as far as possible in the parts that do not affect CUDA support we are working to intoduce.

Reply:
Ok. I will follow your recommendation to remove type annotations from these two functions. I believe this can be done safely because both functions are only used locally. However, I would like to note that this type of code refactoring should be done carefully on code that is already being used in production, because when a type annotation is inadvertently removed, the type of the parameter is, in fact, been changed. Remember that Julia's multiple submission is based on a complex and difficult to understand implementation of the subtyping relation, with no formal safety guarantees (https://fzn.fr/projects/lambdajulia).

Reply:
Ok. I will follow your recommendation to remove type annotations from these two functions. I believe this can be done safely because both functions are only used locally. However, I would like to note that this type of code refactoring should be done carefully on code that is already being used in production, because when a type annotation is inadvertently removed, the type of the parameter is, in fact, been changed. Remember that [Julia's multiple dispatch is based on a complex and difficult to understand implementation of the subtyping relation, with no formal safety guarantees](https://fzn.fr/projects/lambdajulia).

Reply:
Ok. I will follow your recommendation to remove type annotations from these two functions. I believe this can be done safely because both functions are only used locally, and the values of type variables T and N, used in the code, may be retrieved using ndims and eltype functions. However, I would like to note that this type of code refactoring should be done carefully on code that is already being used in production, because when a type annotation is inadvertently removed, the type of the parameter is, in fact, been changed. Remember that [Julia's multiple dispatch is based on a complex and difficult to understand implementation of its own subtyping relation, with no formal safety guarantees](https://fzn.fr/projects/lambdajulia).

Reply:
@decarvalhojunior-fh production code is not development code. Projects that use this package have a Project.toml with a fixed stable release of ImageQuilting.jl We will only release a new version after careful considerations and testing. That is why we have a set of tests to check in every PR. Right now we are still discussing idiomatic issues in Julia code, but hopefully we will soon be able to test the code in the GPU and make sure that it is working as expected. After that we will be able to start thinking about the modular resource-based components and mechanisms.